### PR TITLE
MESH-1084/Fix ambiguous naming in the Compliance Manager

### PR DESCRIPTION
**Pull request by CJP10**

### CODE REVIEW

**Comment by adamdossa** [comment]
@CJP10 @maxsam4 This PR changes some storage keys, so will need a runtime upgrade function I believe.

**Comment by CJP10** [comment]
@Centril can you confirm that the runtime upgrade changes I made will play nice with migrate_map?

File: pallets/compliance-manager/src/lib.rs [hunk 1]
```diff
@@ -257,146 +261,152 @@ decl_module! {
         fn deposit_event() = default;
 
         fn on_runtime_upgrade() -> frame_support::weights::Weight {
+            use frame_support::migration::{StorageIterator, put_storage_value};
             use polymesh_primitives::migrate::migrate_map;
-            migrate_map::<AssetTransferRulesOld>(b"ComplianceManager", b"AssetRulesMap");
 
-            // It's gonna be alot, so lets pretend its 0 anyways.
-            0
+            // ComplianceManager.AssetRulesMap -> ComplianceManager.AssetCompliance
```
Comment (Centril) [inline pallets/compliance-manager/src/lib.rs:267]: This seems right, but I'd prefer `migrate_map_rename` (base function used to impl `migrate_map`) below (using 2 pairs perhaps) to do this in one shot.

**Comment by Centril** [comment]
@CJP10 It seems correct; assuming substrate doesn't care about field names, which it shouldn't in Decode/Encode impls.

File: primitives/src/migrate.rs [hunk 2]
```diff
@@ -53,3 +53,14 @@ pub fn migrate_map<T: Migrate>(module: &[u8], item: &[u8]) {
         .filter_map(|(key, old)| Some((key, old.migrate()?)))
         .for_each(|(key, new)| put_storage_value(module, item, &key, new));
```
Comment (Centril) [inline primitives/src/migrate.rs:54]: This function is equivalent to `migrate_map_rename::<T>(module, item, item);`.

**Review by maxsam4 — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Comment by maxsam4** [comment]
Depends on https://github.com/PolymathNetwork/Polymesh/pull/563 due to settlement stuff.

**Pull request merged**